Release component split (Client, Server, Gateway), update (#2519)
mendersoftware/mender-docs: 302.Release-information/01.Release-schedule/docs.md

---
title: Supported releases
taxonomy:
    category: docs
---

This document outlines the currently supported releases and compatibility between
individual Mender components. To understand the general compatibility policy
of Mender, see the [Compatibility](../../02.Overview/15.Compatibility/docs.md) page.

## Components and LTS releases

> The following long term support (LTS) versions of components have recently
> been released:

Reviewer: I would avoid the word "recently" here, especially since we discussed in the document that we will potentially list the EOL releases too in the table (from next May on).

```suggestion
See the information about the latest long term support (LTS) versions of the Mender components in the following table:
```

* Scarthgap (5.0)
* Kirkstone (4.0)

Each Yocto Project release has a corresponding branch in meta-mender.
Note that you will find other branches in meta-mender but they are not
updated during a future Mender release.

## Mender Artifact format support

The [Mender Artifact format](../../02.Overview/03.Artifact/docs.md) is versioned
to support new features over time. It is used by the Mender Client (to install Artifacts),
Mender Server (to parse and display meta-data, and to generate some types of Artifacts)
and the mender-artifact CLI utility (to create Artifacts).

| Artifact format | Client | Server | mender-artifact |
| --------------- | ------ | ------ | --------------- |
| Artifact v1     | 1.0    | 1.0    | 1.0             |
| Artifact v2     | 1.1    | 1.1    | 2.0             |
| Artifact v3     | 2.0    | 2.0    | 3.0             |

!!! Before you create Artifacts of a new format version, make sure that every Mender
component you use anywhere supports the version of the Artifact format.

## Device API version support

The compatibility between the Mender Server and Client is managed by the

> [Device API](https://docs.mender.io/server-side-api?target=_blank#device-apis)

Reviewer: I know that this is correct, but a newcomer it reads confusing that "Device API" links to "Server-side API" instead of "Device-side API" from the docs menu.

I suggest we link directly into the final destination, with the bonus of pointing directly into the Device API of the Server-side API :melting_face: 

```suggestion
[Device API](https://docs.mender.io/api/#device-apis?target=_blank#device-apis)
```

versions exposed by the Server and used by the Client.
If the Mender Server supports the API version of the Mender Client, they are compatible.

The higher version API contains a mix of old and new API endpoints. For endpoints which
have not changed in the new version, previous version ones are assumed. For example, if

> the Device uses the v2 Device API, it expect

Reviewer: ```suggestion
the Device uses the v2 Device API, it expects
```

>  [this single v2 Deployments endpoint](https://docs.mender.io/api/?target=_blank#device-api-deployments-v2)
>  to be available. For all other endpoints it will use v1.

Reviewer: ```suggestion
[this single v2 Deployments endpoint](https://docs.mender.io/api/?target=_blank#device-api-deployments-v2)
to be available. For all other endpoints it will use v1.
```

| Device API | Server | Gateway | Client |
| ---------- | ------ | ------- | ------ |
| API v1     | 1.0    | N/A     | 1.0    |
| API v2     | 3.0    | 1.0     | 3.0    |

!!! Make sure to upgrade the Server first to support a new API version. Next it is
recommended to update any Gateway components and finally Client.

!!! So far, Mender Server supports all Device API versions ever released. In other words,
a current Mender Server will still work with Mender Client v1.0. However, it is good practice
and required for commercial support to regularly update all your Mender components to

> ensure they all run supported versions. This will prevent issues in the future.

Reviewer: We are missing the `Mender Client subcomponents` section - I replied now on the missing information in the private planning document :ok_hand:

Author: Updated, thank you.
